test/integration/clean/test_clean.py

```diff
@@ -0,0 +1,157 @@
+        for inst in stop_instances:
+            rc, _ = tt_app.stop(inst, '-y')
+            assert rc == 0
+        # Wait for the xlogs (except for the router).
+        # NOTE: This is done here (after stop) because for example 'storage-replica' might
+        # create xlog file only after `tt stop`.
+        xlog_instances = filter(lambda x: 'router' not in x, stop_instances)
+        xlog_files = map(lambda x: tt_app.lib_path(x, '*.snap'), xlog_instances)
+        assert utils.wait_files(5, xlog_files)
+        # And finally the original extra_setup.
+        if extra_setup is not None:
+            extra_setup(tt_app)
+
+    # Prepare the application considering the test parameters.
+    orig_status = ttapp.setup(tt_app, running_targets, post_start,
+                              extra_setup_wrapper, stop_instances, extra_setup)
+
+    # Do clean.
+    rc, out = tt_app.clean(target, *args)
+    assert rc == 0
+
+    # Check the instances.
+    status = tt_app.status()
+    for inst in tt_app.instances:
+        was_running = inst in running_instances
+        assert status[inst]["STATUS"] == orig_status[inst]["STATUS"]
+        if was_running and inst not in stop_instances:
+            assert status[inst]["PID"] == orig_status[inst]["PID"]
+        if inst in target_instances:
+            inst_name = inst.partition(':')[2]
+            msg = f"instance `{inst_name}` must be stopped"
+            if was_running:
+                if inst in stop_instances:
+                    assert msg not in out
+                    # https://github.com/tarantool/tt/issues/735
+                    msg = r"{}...\t\[OK\]".format(inst)
+                    assert len(re.findall(msg, out)) == 1
+                    not os.path.exists(tt_app.log_path(inst, utils.log_file))
+                    not os.path.exists(tt_app.lib_path(inst, utils.initial_snap))
+                    not os.path.exists(tt_app.lib_path(inst, utils.initial_xlog))
+                else:
+                    assert msg in out
+            else:
+                assert f"{inst}...\t[ERR]" in out
+
+
+################################################################
+# Multi-instance
+
+mark_kwargs_multi_inst = dict(
+    app_path='multi_inst_data_app',
+    app_name='app',
+    instances=['router', 'master', 'replica', 'stateboard'],
+)
+
+
+# Auto-confirmation (short option).
+@pytest.mark.slow
+@pytest.mark.tt_app(**mark_kwargs_multi_inst)
+@pytest.mark.parametrize('running_targets, stop_targets', [
+    pytest.param([], [], id="running:none/none"),
+    pytest.param(['app'], [], id="running:all/none"),
+    pytest.param(['app'], ['app'], id="running:all/all"),
+    pytest.param(['app:master'], ['app:master'], id="running:master/master"),
+    pytest.param(['app'], ['app:master'], id="running:all/master"),
+    pytest.param(['app'], ['app:master', 'app:router'], id="running:all/master,router"),
+])
+@pytest.mark.parametrize("extra_setup", [
```

Comment:
I'm not sure I understand your concern. In any test we have "input" and "output" (in a broad sense). Here the presence of the instance script is a part of the test "input" and "output" should not depend on its existence.

> If nothing changed, why it matters to test?

Considering the above this is incorrect statement, because "input" is changed.

Reply:
> the instance script is a part of the test "input" and "output" should not depend on its existence

That's what I find so strange: changes in inputs don't lead to changes in outputs.

Reply:
> That's what I find so strange: changes in inputs don't lead to changes in outputs.

It was the goal of this PR. Previously such change in input did lead to dramatic changes in output because every command mentioned in PR title failed w/o instance script (in case of multi-inst app).